src/controllers/product-controller.ts

```diff
@@ -0,0 +1,72 @@
+import {Context} from 'koa';
+import {
+    retrieveAllProducts,
+    retrieveSpecificProduct,
+    createProduct,
+    updateProduct,
+    deleteProduct,
+    checkout
+} from '../actions/product-db-actions';
+import {creationSchema, updateSchema, checkoutSchema} from '../validation-schemas/product-request-schemas';
+import {logger} from '../logger/logger';
+
+export const getAllProductsRequest = async (ctx: Context) => {
+    ctx.body = await retrieveAllProducts();
+};
+
+export const getSpecificProductRequest = async (ctx: Context) => {
+    const product = await retrieveSpecificProduct(ctx.params.id);
+    if (!product)
+        ctx.throw(404, 'Product not found!');
```

Comment:
`ctx.throw` is not ending this function from running....

Reply:
same in the next controllers

Reply:
are you sure? I have read that it's equivalent to 
`const err = new Error('blabla');
err.status = ...;
err.expose = true;
throw err;`

But I added return statements anyway